src/EnergyPlus/DataGlobalConstants.cc

```diff
@@ -225,150 +225,148 @@ std::string GetResourceTypeChar(ResourceType const ResourceTypeNum)
-            return "Steam";
-
-        } else if (SELECT_CASE_var == ResourceType::DistrictCooling) {
-            return "DistrictCooling";
-
-        } else if (SELECT_CASE_var == ResourceType::DistrictHeating) {
-            return "DistrictHeating";
-
-        } else if (SELECT_CASE_var == ResourceType::ElectricityProduced) {
-            return "ElectricityProduced";
-
-        } else if (SELECT_CASE_var == ResourceType::ElectricityPurchased) {
-            return "ElectricityPurchased";
-
-        } else if (SELECT_CASE_var == ResourceType::ElectricitySurplusSold) {
-            return "ElectricitySurplusSold";
-
-        } else if (SELECT_CASE_var == ResourceType::ElectricityNet) {
-            return "ElectricityNet";
-
-        } else if (SELECT_CASE_var == ResourceType::SolarWater) {
-            return "SolarWater";
-
-        } else if (SELECT_CASE_var == ResourceType::SolarAir) {
-            return "SolarAir";
-
-        } else if (SELECT_CASE_var == ResourceType::SO2) {
-            return "SO2";
-
-        } else if (SELECT_CASE_var == ResourceType::NOx) {
-            return "NOx";
-
-        } else if (SELECT_CASE_var == ResourceType::N2O) {
-            return "N2O";
-
-        } else if (SELECT_CASE_var == ResourceType::PM) {
-            return "PM";
-
-        } else if (SELECT_CASE_var == ResourceType::PM2_5) {
-            return "PM2.5";
-
-        } else if (SELECT_CASE_var == ResourceType::PM10) {
-            return "PM10";
-
-        } else if (SELECT_CASE_var == ResourceType::CO) {
-            return "CO";
-
-        } else if (SELECT_CASE_var == ResourceType::CO2) {
-            return "CO2";
-
-        } else if (SELECT_CASE_var == ResourceType::CH4) {
-            return "CH4";
-
-        } else if (SELECT_CASE_var == ResourceType::NH3) {
-            return "NH3";
-
-        } else if (SELECT_CASE_var == ResourceType::NMVOC) {
-            return "NMVOC";
-
-        } else if (SELECT_CASE_var == ResourceType::Hg) {
-            return "Hg";
-
-        } else if (SELECT_CASE_var == ResourceType::Pb) {
-            return "Pb";
-
-        } else if (SELECT_CASE_var == ResourceType::NuclearHigh) {
-            return "Nuclear High";
-
-        } else if (SELECT_CASE_var == ResourceType::NuclearLow) {
-            return "Nuclear Low";
-
-        } else if (SELECT_CASE_var == ResourceType::WaterEnvironmentalFactors) {
-            return "WaterEnvironmentalFactors";
-
-        } else if (SELECT_CASE_var == ResourceType::CarbonEquivalent) {
-            return "Carbon Equivalent";
-
-        } else if (SELECT_CASE_var == ResourceType::Source) {
-            return "Source";
-
-        } else if (SELECT_CASE_var == ResourceType::PlantLoopHeatingDemand) {
-            return "PlantLoopHeatingDemand";
-
-        } else if (SELECT_CASE_var == ResourceType::PlantLoopCoolingDemand) {
-            return "PlantLoopCoolingDemand";
-
-        } else {
-            return "Unknown";
-        }
+    switch (ResourceTypeNum) {
```

Comment:
This should not be a function at all, it should be a `constexpr std::array<std::string_view>`.